packages/conjure-client/src/fetchBridge/__tests__/fetchBridgeTests.ts

```diff
@@ -477,6 +485,46 @@ describe("FetchBridgeImpl", () => {
                 }),
         ).toThrowError(new Error("At least one user agent must be provided"));
     });
+
+    describe("sends correct Fetch-User-Agent", () => {
+        const request: IHttpEndpointOptions = {
+            endpointName: "a",
+            endpointPath: "a/{var}/b",
+            method: "GET",
+            pathArguments: ["val"],
+            queryArguments: {},
+            responseMediaType: MediaType.APPLICATION_JSON,
+        };
+
+        beforeEach(() => {
+            // Mocking all requests as we're only interested in request headers
+            fetchMock.mock(() => true, {});
+        });
+
+        it("for user agent", async () => {
+            const fetchBridge = new FetchBridge({ baseUrl, token, fetch: undefined, userAgent });
+
+            await fetchBridge.callEndpoint(request);
+
+            verifyFetchUserAgent();
+        });
+
+        it("for user agent array", async () => {
+            const fetchBridge = new FetchBridge({ baseUrl, token, fetch: undefined, userAgent: [userAgent] });
+
+            await fetchBridge.callEndpoint(request);
+
+            verifyFetchUserAgent();
+        });
+
+        it("for user agent supplier array", async () => {
+            const fetchBridge = new FetchBridge({ baseUrl, token, fetch: undefined, userAgent: [() => userAgent] });
+
+            await fetchBridge.callEndpoint(request);
+
+            verifyFetchUserAgent();
+        });
```

Comment:
Can we have a test that modifies the dynamically provided user agent and verifies that the fetch header changes?

Otherwise it would be easy for a future modification to e.g. locally cache the userAgent and this would break.